pkg/util/validation/exporter/ring.go

```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package exporter
+
+import (
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/flagext"
+	"github.com/grafana/dskit/kv"
+	"github.com/grafana/dskit/netutil"
+	"github.com/grafana/dskit/ring"
+
+	util_log "github.com/grafana/mimir/pkg/util/log"
+)
+
+const (
+	// ringKey is the key under which we store the overrides-exporter's ring in the KVStore.
+	ringKey = "overrides-exporter"
+
+	// ringNumTokens is how many tokens each overrides-exporter should have in the ring.
+	// Overrides-exporters use a ring for discovery only, so one token is enough.
+	ringNumTokens = 1
+
+	// ringAutoForgetUnhealthyPeriods is how many consecutive timeout periods an unhealthy instance
+	// in the ring will be automatically removed after.
+	ringAutoForgetUnhealthyPeriods = 4
+)
+
+var instanceStateFilter = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
+
+type RingConfig struct {
```

Comment:
I think this makes the 4th copy of basically the same code:

* Distributor
* Compactor
* Query-scheduler
* Now this.

Maybe we can share some form of it between them?

Reply:
I agree, but I would suggest doing it in a separate PR.